lmp/tokenizer/_base_dict_tokenizer.py

```diff
@@ -323,9 +360,21 @@ def batch_encode(
                 longer than `max_seq_len` and padded to `max_seq_len` when
                 individual sequence length is shorter than `max_seq_len`.
 
+        Raises:
+            TypeError:
+                When `batch_sequences` is not instance of `List[str]`.
+                When `max_seq_len` is not instance of `int`.
+
         Returns:
             Batch of token ids encoded from `batch_sequence`.
         """
+        # Type check.
+        if not isinstance(batch_sequences, List):
+            raise TypeError(
+                '`batch_sequences` must be instance of `List[str]`.')
```

Comment:
new line before right parenthesis

Reply:
First do `batch_sequences = list(batch_sequences)`, then use `for` loop to check each element in `batch_sequences` is an instance of `str`